QModManager/Patcher.cs

```diff
@@ -120,22 +124,98 @@ internal static void LoadMods()
+                            // Reason we check if its already in the list is because there is going to be at least 1 duplicate
+                            if(!erroredMods.Contains(modSortingChain[j]))
+                                erroredMods.Add(modSortingChain[j]);
+                        }
+                    }
+
+                    sortingErrorLoops.Add(loop);
+                }
+            }
+
+            if(sortingErrorLoops.Count != 0)
+            {
+                Console.WriteLine("QMOD ERROR: There was en error while sorting the following mods!");
+                Console.WriteLine("Please check the 'LoadAfter' and 'LoadBefore' properties of these mods!\n");
+
+                foreach (List<QMod> list in sortingErrorLoops)
+                {
+                    string outputStr = "";
+
+                    foreach(QMod mod in list)
+                    {
+                        // Remove it from list to prevent it from being loaded
+                        if(sortedMods.Contains(mod))
+                            sortedMods.Remove(mod);
+
+                        outputStr += mod.Id + " -> ";
+                    }
+
+                    outputStr = outputStr.Substring(0, outputStr.Length - 4);
+                    Console.WriteLine(outputStr);
+                }
+            }
+
+            // Check if all mods have dependencies present
+            Dictionary<QMod, List<string>> missingDependenciesByMod = new Dictionary<QMod, List<string>>();
+        
+            foreach(QMod mod in foundMods)
+            {
+                List<QMod> presentDependencies = GetDependenciesPresent(mod);
+                List<string> missingDependencies = GetDependenciesMissing(mod, presentDependencies);
+
+                if (missingDependencies.Count != 0)
+                {
+                    missingDependenciesByMod.Add(mod, missingDependencies);
+
+                    // Add this mod to the list of errored mods
+                    if (!erroredMods.Contains(mod))
+                        erroredMods.Add(mod);
+                }
+            }
 
-                    // Last element of the list will always be a duplicate, so no point to include it
-                    for (int y = 0; y < modSortingChain.Count - 1; y++)
+            // There are missing dependencies! Output them!
+            if(missingDependenciesByMod.Count != 0)
+            {
+                Console.WriteLine("QMOD ERROR: The following mods were not loaded due to missing dependencies!\n");
+
+                foreach(var entry in missingDependenciesByMod)
+                {
+                    // Remove the mod from the sortedMods list to stop it from loading
+                    if(sortedMods.Contains(entry.Key))
+                        sortedMods.Remove(entry.Key);
+
+                    // Build the string to be displayed for this mod
+                    string str = entry.Key.DisplayName + ": Missing ";
```

Comment:
This format: `Mod1: Missing Mod2` isn't the best one IMO, maybe change it to `Mod1 (missing: Mod2)`